hydrolib/hydromt_delft3dfm/dflowfm.py

```diff
@@ -1382,58 +1660,126 @@ def write(self):  # complete model
-        if not self._write:
-            raise IOError("Model opened in read-only mode")
-        self.staticmaps.raster.to_mapstack(join(self.root, "dflowfm"))
+    def write_auxmaps(self) -> None:
+        """Write auxmaps as tif files in auxmaps folder and update initial fields"""
+        # Global parameters
+        auxroot = join(self.root, "auxmaps")
+        inilist = []
+        paramlist = []
+        self.logger.info(f"Writing auxmaps files to {auxroot}")
+
+        def _prepare_inifields(da_dict, da):
+            # Write tif files
+            name = da_dict["name"]
+            type = da_dict["initype"]
+            interp_method = da_dict["interpolation"]
+            locationtype = da_dict["locationtype"]
+            _fn = join(auxroot, f"{name}.tif")
+            if np.isnan(da.raster.nodata):
+                da.raster.set_nodata(-999)
+            da.raster.to_raster(_fn)
+            # Prepare dict
+            if interp_method == "constant" or interp_method == "triangulation":
+                inidict = {
+                    "quantity": name,
+                    "dataFile": f"../auxmaps/{name}.tif",
+                    "dataFileType": "GeoTIFF",
+                    "interpolationMethod": interp_method,
+                    "operand": "O",
+                    "locationType": locationtype,
+                }
+            else:  # averaging
+                inidict = {
+                    "quantity": name,
+                    "dataFile": f"../auxmaps/{name}.tif",
+                    "dataFileType": "GeoTIFF",
+                    "interpolationMethod": "averaging",
+                    "averagingType": interp_method,
+                    "operand": "O",
+                    "locationType": locationtype,
+                }
+            if type == "initial":
+                inilist.append(inidict)
+            elif type == "parameter":
+                paramlist.append(inidict)
+
+        # Only write auxmaps that are listed in self._AUXMAPS, rename tif on the fly
+        for name, ds in self._auxmaps.items():
+            if isinstance(ds, xr.DataArray):
+                if name in self._AUXMAPS:
+                    _prepare_inifields(self._AUXMAPS[name], ds)
+                    # update config if frcition
+                    if "frictype" in self._AUXMAPS[name]:
+                        self.set_config(
+                            "physics.UniFrictType", self._AUXMAPS[name]["frictype"]
+                        )
+            elif isinstance(ds, xr.Dataset):
+                for v in ds.data_vars:
+                    if v in self._AUXMAPS:
+                        _prepare_inifields(self._AUXMAPS[v], ds[v])
+                        # update config if frcition
+                        if "frictype" in self._AUXMAPS[name]:
+                            self.set_config(
+                                "physics.UniFrictType", self._AUXMAPS[name]["frictype"]
+                            )
+        # rewrite config
```

Comment:
Raise value error if both waterlevel and waterdepth are specified self._auxmaps.items().name. Only one can be used.